pgxn/neon/file_cache.c

```diff
@@ -617,31 +617,34 @@ lfc_evict(NRelFileInfo rinfo, ForkNumber forkNum, BlockNumber blkno)
 	/* remove the page from the cache */
 	entry->bitmap[chunk_offs >> 5] &= ~(1 << (chunk_offs & (32 - 1)));
 
-	/*
-	 * If the chunk has no live entries, we can position the chunk to be
-	 * recycled first.
-	 */
-	if (entry->bitmap[chunk_offs >> 5] == 0)
+	if (entry->access_count == 0)
```

Comment:
>  with one further improvement suggestion in case you're interested in doing that.

Sorry, what suggestion do you mean?

> Should we also have such a check in the relinking stage of lfc_read/lfc_write to make sure that fully cleared chunks are always linked at the head of the LRU, or would you consider that too expensive?

Sorry, but `lfc_evict` is the only place when we actually clear entry bitmap.
Both lfc_read and lfc_write can only add new elements to this map, they never can make it empty.
That is why O do not think that there is any sense in checking state of bitmap here.

Yes, if we refused to move entry in `lfc_eict` because it is accessed at this moment, then for example `lfc_read` can access empty chunk. But probability of it is expected be very small, so I do not think that we should add check if chunk is empty here.